### PR TITLE
WIP: Very preliminary cmake support

### PR DESCRIPTION
**Pull request by brianretford**
cmake -G Ninja -B cmb
cmake --build cmb --target PMLCEltwise

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+#####
+# Easylogging
+####
+FetchContent_Declare(
+  easyloggingpp
+  FETCHCONTENT_QUIET OFF
+  URL https://github.com/amrayn/easyloggingpp/archive/v9.96.7.tar.gz
+  URL_HASH SHA256=237c80072b9b480a9f2942b903b4b0179f65e146e5dcc64864dc91792dedd722
+)
+FetchContent_GetProperties(easloggingpp)
```
Comment (flaub) [inline CMakeLists.txt:70]: typo: easyloggingpp

File: pmlc/dialect/eltwise/ir/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+add_mlir_dialect_library(PMLCEltwise
+        dialect.cc
+        ops.cc
+
+        ADDITIONAL_HEADER_DIRS
+
+        DEPENDS
+                MLIROpAsmInterfaceIncGen
+                PMLCeltwiseDialectGen
```
Comment (flaub) [inline pmlc/dialect/eltwise/ir/CMakeLists.txt:15]: I'd say `PMLCEltwiseDialectGen`

File: pmlc/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+function(pmlc_add_mlir_interfaces namespace)
+  set(LLVM_TARGET_DEFINITIONS interfaces.td)
+  mlir_tablegen(interfaces.h.inc -gen-op-interface-decls)
+  mlir_tablegen(interfaces.cc.inc -gen-op-interface-defs)
+  add_public_tablegen_target(PMLC${namespace}InterfacesGen)
+  add_dependencies(mlir-headers PMLC${namespace}InterfacesGen)
+endfunction()
+
+add_subdirectory(util)
+add_subdirectory(dialect)
```
Comment (flaub) [inline pmlc/CMakeLists.txt:20]: add newline at end

File: pmlc/dialect/eltwise/ir/CMakeLists.txt (hunk 2 above)
Comment (brianretford) [reply pmlc/dialect/eltwise/ir/CMakeLists.txt:15]: kk, involves some annoying string processing in the macros is all - uses the tablegen file name to populate that. Ideally the rule would be explicit and have a name and a list of tablegen files etc. I can at least make the name / td file separate

Comment (brianretford) [reply pmlc/dialect/eltwise/ir/CMakeLists.txt:15]: Derp, that was how the old one worked, this is easy nm

File: vendor/openvino/openvino.BUILD [hunk 4]
```diff
@@ -1,7 +1,7 @@
-load("@rules_cc//cc:defs.bzl", "cc_library")
-load("@com_intel_plaidml//bzl:template.bzl", "template_rule")
+LOAD("@RULES_CC//CC:DEFS.BZL", "CC_LIBRARY")
+LOAD("@COM_INTEL_PLAIDML//BZL:TEMPLATE.BZL", "TEMPLATE_RULE")
 
-package(default_visibility = ["//visibility:public"])
+PACKAGE(DEFAULT_VISIBILITY = ["//VISIBILITY:PUBLIC"])
```
Comment (flaub) [inline vendor/openvino/openvino.BUILD:4]: Not sure if this was intended, looks like this file got capitalized somehow

**Pull request closed without merging**